### PR TITLE
refactor: utils repository

### PR DESCRIPTION
**Pull request by 0xtekgrinder**

### CODE REVIEW

File: scripts/Constants.s.sol [hunk 1]
```diff
@@ -7,18 +7,8 @@ import "contracts/utils/Constants.sol";
...
 
-address constant ACCESS_CONTROL_MANAGER = 0x5bc6BEf80DA563EBf6Df6D6913513fa9A7ec89BE;
-address constant CHAIN_AGEUR = 0x1a7e4e63778B4f12a199C062f3eFdD288afCBce8;
-address constant GOVERNOR = 0xdC4e6DFe07EFCa50a197DF15D9200883eF4Eb1c8;
-address constant GUARDIAN = 0x0C2553e4B9dFA9f83b1A6D3EAB96c4bAaB42d430;
-address constant PROXY_ADMIN = 0x1D941EF0D3Bba4ad67DBfBCeE5262F4CEE53A32b;
-address constant PROXY_ADMIN_GUARDIAN = 0xD9F1A8e00b0EEbeDddd9aFEaB55019D55fcec017;
-address constant TREASURY_EUR = 0x8667DBEBf68B0BFa6Db54f550f41Be16c4067d60;
-address constant TREASURY_USD = 0xf8588520E760BB0b3bDD62Ecb25186A28b0830ee;
 address constant IMMUTABLE_CREATE2_FACTORY_ADDRESS = 0x0000000000FFe8B47B3e2130213B802212439497;
```
Comment (GuillaumeNervoXS) [inline scripts/Constants.s.sol:10]: Are we sure this is the address on every chain?

Comment (0xtekgrinder) [reply scripts/Constants.s.sol:10]: the immutable factory is the same on every chain

File: scripts/DeploySavingsNoImplem.s.sol [hunk 2]
```diff
@@ -36,8 +36,8 @@ contract DeploySavingsNoImplem is Utils {
         // No need to deploy the implementation here
         // TODO: update addresses based on deployment
         address agToken = 0x0000206329b97DB379d5E1Bf586BbDB969C63274;
```
Comment (GuillaumeNervoXS) [inline scripts/DeploySavingsNoImplem.s.sol:38]: can be changed also to query utils (agEUR or agUSD)

File: scripts/utils/VanityAddress.s.sol [hunk 3]
```diff
@@ -1,62 +1,24 @@
...
-        }
+        uint256 initInt = 0;
+        uint256 iterations = 3000000;
 
-        console.log("found ", found);
-        console.log("i ", i);
-        console.logBytes32(bytes32(abi.encodePacked(DEPLOYER, abi.encodePacked(uint96(i)))));
-        console.log("computedAddress ", computedAddress);
+        (address computed, uint256 found) = minePrefix(initCode, DEPLOYER, 0x002222, initInt, iterations);
+        console.log("Computed: ", computed);
```
Comment (GuillaumeNervoXS) [inline scripts/utils/VanityAddress.s.sol:20]: We should write the result in JSON_VANITY_PATH json

File: test/Fixture.sol [hunk 4]
```diff
@@ -57,6 +58,10 @@ contract Fixture is Transmuter {
         dylan = vm.addr(4);
         sweeper = address(uint160(uint256(keccak256(abi.encodePacked("sweeper")))));
 
+        governor = _chainToContract(CHAIN_ETHEREUM, ContractType.GovernorMultisig);
```
Comment (GuillaumeNervoXS) [inline test/Fixture.sol:61]: Let's set a chain variable so we only need to change in one place

File: scripts/DeploySavingsNoCreate2.s.sol [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
-/// constants defined for the chain of your choice.
+/// @dev To deploy on a different chain, just replace the chainId and be sure the sdk has the required addresses
 /// @dev This is a vanilla deployment file to easily and rapidly deploy a savings implementation at a random address
 contract DeploySavingsNoCreate2 is Utils {
     function run() external {
         // TODO: make sure that deployer has a 1 agEUR (=1e18) balance
-        // TODO: check the import of the constants file if it corresponds to the chain you're deploying on
+        // TODO: change the chainId
+        uint256 chainId = CHAIN_ETHEREUM;
         uint256 deployerPrivateKey = vm.deriveKey(vm.envString("MNEMONIC_FORK"), "m/44'/60'/0'/0/", 0);
```
Comment (GuillaumeNervoXS) [inline scripts/DeploySavingsNoCreate2.s.sol:19]: we can also use the CHAIN_SOURCE

File: scripts/DeploySavingsNoImplem.s.sol [hunk 6]
```diff
@@ -16,15 +16,15 @@ import { ImmutableCreate2Factory } from "./utils/TransmuterDeploymentHelper.s.so
...
+/// @dev To deploy on a different chain, just replace the chainId and be sure the sdk has the required addresses
 contract DeploySavingsNoImplem is Utils {
     using stdJson for string;
     using strings for *;
 
     function run() external {
         // TODO: make sure that deployer has a 1 stablecoin (=1e18) balance
-        // TODO: check the import of the constants file if it corresponds to the chain you're deploying on
+        // TODO: change the chainId
+        uint256 chainId = CHAIN_ETHEREUM;
```
Comment (GuillaumeNervoXS) [inline scripts/DeploySavingsNoImplem.s.sol:27]: same

File: test/utils/Transmuter.sol [hunk 7]
```diff
@@ -18,6 +18,8 @@ import { Swapper } from "contracts/transmuter/facets/Swapper.sol";
 import "contracts/utils/Errors.sol";
 import { DummyDiamondImplementation } from "../../scripts/generated/DummyDiamondImplementation.sol";
 
+import { CHAIN_SOURCE } from "../../scripts/Constants.s.sol";
```
Comment (GuillaumeNervoXS) [inline test/utils/Transmuter.sol:21]: useless no?

File: scripts/utils/VanityAddress.s.sol [hunk 8]
```diff
@@ -1,62 +1,34 @@
...
+        console.log("Found: ", found);
 
-        console.log("found ", found);
-        console.log("i ", i);
-        console.logBytes32(bytes32(abi.encodePacked(DEPLOYER, abi.encodePacked(uint96(i)))));
-        console.log("computedAddress ", computedAddress);
+        // write result to json vanity path
+        json = "";
+        vm.serializeUint(json, "init", found);
```
Comment (GuillaumeNervoXS) [inline scripts/utils/VanityAddress.s.sol:29]: Did you test that it was indeed writing well ion the file?

Comment (0xtekgrinder) [reply scripts/utils/VanityAddress.s.sol:29]: yep 👍 it works good

**Review by GuillaumeNervoXS — APPROVED** [review]

**Pull request merged**